src/components/Button.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <button
+    class="f-button"
+    type="button"
+    @click="onClick">
+    {{text}}
+  </button>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component
+export default class Button extends Vue {
+  // region public static methods
+  // endregion
+
+  // region private static methods
+  // endregion
+
+  // region public members
+  @Prop(String)
+  public text!: string;
+
+  @Prop(Function)
+  public onClick!: () => void;
```

Comment:
why do we need the '!' here?

Reply:
Because we ensure to the class that onClick is set.